lib/Reader/readerir.cpp

```diff
@@ -460,36 +463,21 @@ void GenIR::readerPostVisit() {
 }
 
 void GenIR::readerPostPass(bool IsImportOnly) {
-  if (JitContext->Options->DoInsertStatepoints) {
-
-    // Precise GC using statepoints cannot handle aggregates that contain
-    // managed pointers yet. So, check if this function deals with such values
-    // and fail early. (Issue #33)
 
-    for (const Argument &Arg : Function->args()) {
-      if (isManagedAggregateType(Arg.getType())) {
-        throw NotYetImplementedException(
-            "NYI: Precice GC for Managed-Aggregate values");
-      }
-    }
-
-    for (const BasicBlock &BB : *Function) {
-      for (const Instruction &Instr : BB) {
-        if (isManagedAggregateType(Instr.getType())) {
-          throw NotYetImplementedException(
-              "NYI: Precice GC for Managed-Aggregate values");
-        }
+  if (JitContext->Options->DoInsertStatepoints) {
+    SmallVector<Value *, 4> EscapingLocs;
+    GcFuncInfo->getEscapingLocations(EscapingLocs);
```

Comment:
I think we want a general capability to handle multiple escapes, no matter what the GC situation is.